src/scripts/main.js

```diff
@@ -1,3 +1,226 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+
+let headers, tbody, rows;
+
+const init = () => {
+  headers = [...body.querySelectorAll('th')];
+  tbody = body.querySelector('tbody');
+  rows = [...tbody.querySelectorAll('tr')];
+};
```

Comment:
The `init` function is called only once at the start of the script and then once again after a new row is added to the table. It would be more efficient to update `rows` only when a new row is added, instead of querying the DOM again.